zcommon/src/org/zkoss/lang/Library.java

```diff
@@ -154,13 +150,13 @@ public static int getIntProperty(String key, int defVal) {
 		if (key == null) throw new NullPointerException();
 		if (key.length() == 0) throw new IllegalArgumentException();
 
+		List<String> old = _props.get(key);
 		synchronized (_props) {
-			List<String> old = _props.get(key);
 			if (old == null)
 				old = new LinkedList<String>();
 			old.add(value);
-			return _props.put(key, old);
 		}
+		return _props.put(key, old);
```

Comment:
It is better to remain the original code.
The fetch-create-put action should be atomic. Or every thread will replace others.

Reply:
ok